### PR TITLE
Fostrich responsive design improvements

### PR DESCRIPTION
**Pull request by frost-ostrich**
# Description

- [x] Improve metrics in dashboard, responsiveness using flex wrap
- [x] Add support for SVG react components using `vite-plugin-svgr`
- [x] Change various `png` for newest `svg` for easier `fill` & `stroke` color changes 
- [x] Solve various console.log errors on app
- [x] `All` dashboard type tab is now `TRV`
- [x] Remove `width` in terms of `vh`, now all dashboard components adjust to parent width 
- [x] menu icon on figma is 24x24px, on phone it's 40x40px
- [x] when expanded, the TEMPLE logo is centred rather than on the left, and the 'connect' tab is not 'light' colour matching the light icon
- [x] the top navigation options don't fit on the page and should be scrollable to the left but are not
- [x] the padding on the left and right are too wide, the figma design paddings are 20px on either side
- [x] 'Treasury Reserves Vault' title is 36 px but should be 24px as per Figma
- [x] spacing between 'Treasury Reserves Vault' and the 'total market value' filter is too wide
- [x] 'total market value' filter should be full width
- [x] selected graph time filter (1D, 1W, 1M, 1Y) should be underlined https://github.com/TempleDAO/temple/pull/908
- [x] ~~graph is overlapping strategy key for the graph~~
- [x] the metric fonts are too small - on figma they are min 16px
- [x] not enough space between the metrics and the 'transaction history' section
- [x] 'transaction history' font is too big
- [x] selected filter for the transaction history should be 'light' coloured rather than 'base'
- [x] page selections should be at the bottom of the table rather than at the top
- [x] Transaction history table mobile view
- [x] Use single expandable row on transaction history click 
- [x] the top nav for 'trv', 'ramos' etc are 16px in Figma, in dev it's currently 19.2px
- [x] 'etherscan' link next to TRV title 
- [x] too much space between the description and the graph's 'total market value' filter - should be 30px
- [x] the fonts in the box and the 'transaction history' title are H4 on Figma (18px and a font style called Megan) 
- [x] the fonts used for the 'transaction history' transaction period filters and the chart filters are using different fonts, use same as the chart's time period filter's font (should 16px caviar dreams bold weight)
- [x] filter icon should be closer to the column heading for the table - should be 5px
- [x] set `trade` dashboard page as default/first
- [x] the chart 'total market value' filter is 20px in uat but should be 12px
- [x] 'transaction history' heading is still 24px but should be 18px
- [x] spacing for 'transaction history' should be larger at the top and smaller at the bottom so that it's closer to the table filters
- [x] In the table expansion row, there is a slight misalignment for the 'type' column being a little higher horizontally than the rest? and the etherscan icon is vertically misaligned to the dropdown arrow

*Improve navigation bar on both mobile and desktop*
- [x] Selected tab svg color change
- [x] Wallet connect/disconnect css updates
- [x] Clean src 
- [x] Added Navigation menus as react state
- [x] Added disconnect & wallet dropdown menu on nav bar
- [x] Added `useGeoBlocked` hook

### CODE REVIEW

**Comment by nichosystem** [comment]
Deploy preview for _core_ ready!

✅ Preview
https://core-6zphmvxtv-templedao.vercel.app
https://pr-903-preview.templedao.link

Built with commit 3e7a99bf0879c931096486d2018464f783346d4c.
This pull request is being automatically deployed with [vercel-action](https://github.com/marketplace/actions/vercel-action)

File: apps/dapp/src/components/Pages/Core/DappPages/Dashboard/Metrics/index.tsx [hunk 1]
```diff
@@ -84,41 +48,39 @@ const DashboardMetrics = ({ dashboardType }: DashboardMetricsProps) => {
...
+  const desktopView = (sourceData: ArrangedDashboardMetrics) => (
     <>
       <MetricsContainer>
-        {sourceData!.metrics.map((row, idx) => (
-          <MetricsRow key={idx}>
-            {row.map((metric, idx) => (
+        <MetricsRow>
+          {sourceData.metrics.map((row) =>
+            row.map((metric, idx) => (
               <Metric key={idx}>
```
[the remark this answers is not in the document]
Comment (marshall2112) [reply apps/dapp/src/components/Pages/Core/DappPages/Dashboard/Metrics/index.tsx:57]: Good catch, will make a note for the fixes post release.

File: apps/dapp/src/components/Pages/Core/DappPages/Dashboard/Metrics/index.tsx [hunk 2]
```diff
@@ -84,41 +48,39 @@ const DashboardMetrics = ({ dashboardType }: DashboardMetricsProps) => {
...
   );
 
-  const desktopView = (sourceData: Nullable<ArrangedDashboardMetrics>) => (
+  const desktopView = (sourceData: ArrangedDashboardMetrics) => (
     <>
       <MetricsContainer>
-        {sourceData!.metrics.map((row, idx) => (
-          <MetricsRow key={idx}>
-            {row.map((metric, idx) => (
+        <MetricsRow>
```
Comment (marshall2112) [inline apps/dapp/src/components/Pages/Core/DappPages/Dashboard/Metrics/index.tsx:54]: The original design had multiple rows and now it's all in single rows.

File: apps/dapp/src/components/Pages/Core/DappPages/Dashboard/Metrics/index.tsx [hunk 3]
```diff
@@ -84,41 +48,39 @@ const DashboardMetrics = ({ dashboardType }: DashboardMetricsProps) => {
...
-      </MetricsContainer>
-      <MetricsContainer>
-        {sourceData!.smallMetrics.map((row, idx) => (
-          <MetricsRow key={idx}>
-            {row.map((metric, idx) => (
+            ))
+          )}
+        </MetricsRow>
+        <MetricsRow>
+          {sourceData.smallMetrics.map((row) =>
```
Comment (marshall2112) [inline apps/dapp/src/components/Pages/Core/DappPages/Dashboard/Metrics/index.tsx:65]: Screenshot 2023-11-16 at 7 33 42 AM

The alignment seems off for this row.

**Review by marshall2112 — CHANGES_REQUESTED** [review]

File: apps/dapp/package.json [hunk 4]
```diff
@@ -94,7 +94,8 @@
     "tsconfig-paths-webpack-plugin": "^3.5.2",
     "typechain": "^8.1.1",
     "typescript": "^4.9.4",
-    "vite": "^2.7.2"
+    "vite": "^2.7.2",
+    "vite-plugin-svgr": "^4.2.0"
```
Comment (marshall2112) [inline apps/dapp/package.json:98]: What command did you use to add this? I don't see any root level `yarn.lock` changes I would expect. In the monorepo we have to use a specific command to add it such as 

`yarn workspace @temple/dapp add vite-plugin-svgr`

**Review by marshall2112 — CHANGES_REQUESTED** [review]
One thing I see, when "Dashboard" is selected, the left nav shifts slightly. I am not sure yet why.

File: apps/dapp/package.json (hunk 4 above)
Comment (frost-ostrich) [reply apps/dapp/package.json:98]: I think I've discard the changes on that file by mistake, thx for pointing out

File: apps/dapp/src/components/Layouts/V2Layout/index.tsx [hunk 5]
```diff
@@ -1,21 +1,96 @@
...
+      linkTo: V2DashboardLocPaths.Legacy,
+      Logo: Restore,
+      selected: V2DashboardLocPaths.Legacy === loc.pathname,
+    },
+  ]);
+
+  const onSelectMenuNavItems = async (mi: MenuNavItem) => {
+    await setMenuNavItems((prevState) => {
+      const newState = prevState.map((m) => {
+        if (m == mi) {
```
Comment (marshall2112) [inline apps/dapp/src/components/Layouts/V2Layout/index.tsx:77]: Nit: Not a big deal, certainly not a blocker, but would prefer to name these in a more descriptive way such as "previouslySelectedMenuItem" and "selectedMenuItem".

File: apps/dapp/src/components/Layouts/V2Layout/Nav/NavLinks.tsx [hunk 6]
```diff
@@ -62,109 +50,78 @@ const NavLinks = ({ isNavCollapsed = false, onClickHandler }: NavLinksProps) =>
...
-                          e.preventDefault();
-                        }
-                      }}
-                    >
-                      {
-                        <NavLinkText>
-                          <TruncatedAddress address={walletAddress || ''} />
-                        </NavLinkText>
+      <WalletContainer>
+        {/* // TODO: We also need an "Off" or disconnected version of this button */}
```
Comment (marshall2112) [inline apps/dapp/src/components/Layouts/V2Layout/Nav/NavLinks.tsx:84]: Comment no longer relevant as this issue has been fixed.

**Review by marshall2112 — APPROVED** [review]
Just two small things but nothing blocking, LGTM.

**Review by marshall2112 — APPROVED** [review]

**Pull request merged**